hs_geographic_feature_resource/models.py

```diff
@@ -0,0 +1,390 @@
+from django.db import models
+from mezzanine.pages.page_processors import processor_for
+from hs_core.models import BaseResource, ResourceManager, resource_processor, CoreMetaData, AbstractMetaDataElement
+from mezzanine.pages.models import Page
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from django.contrib.contenttypes import generic
+
+# Create your models here.
+
+
+class OriginalFileInfo(AbstractMetaDataElement):
+
+    term = 'OriginalFileInfo'
+
+    fileTypeEnum=(  (None, 'Unknown'),
+                    ("SHP", "ESRI Shapefiles"),
+                    ("ZSHP", "Zipped ESRI Shapefiles"),
+                    ("KML", "KML"),
+                    ("KMZ", "KMZ"),
+                    ("GML", "GML"),
+                    ("SQLITE", "SQLite")
+                 )
+    fileType=models.TextField(max_length=128, choices=fileTypeEnum, default=None)
+    baseFilename=models.TextField(max_length=256, null=False, blank=False)
+    fileCount=models.IntegerField(null=False, blank=False, default=0)
+
+    filenameString = models.TextField(max_length=2048, null=True, blank=True)
+
+    class Meta:
+        # OriginalFileInfo element is not repeatable
+        unique_together = ("content_type", "object_id")
+
+    @classmethod
+    def create(cls, **kwargs):
+
+        for limit in ['fileType', 'baseFilename', 'fileCount']:
+            if not limit in kwargs:
+                raise ValidationError("For OriginalFileInfo meta, one or more attribute is missing.")
+
+        metadata_obj = kwargs['content_object']
+        ori_file_info = OriginalFileInfo.objects.create(fileType=kwargs['fileType'], baseFilename=kwargs['baseFilename'],
+                                                  fileCount=kwargs['fileCount'],
+                                                 content_object=metadata_obj,)
+        # save filenameString info
+        for key, value in kwargs.iteritems():
+            if key in ('filenameString'):
+                setattr(ori_file_info, key, value)
+                ori_file_info.save()
```

Comment:
More information:

The second for loop isn't doing anything at all, because you already created the object with the same fields. You can remove this entirely.

But, again, none of this is necessary at all. This is all basically passing the same parameters to `OriginalFileInfo.objects.create()` and isn't even necessary _at all_. If we want to keep this to be consistent, it doesn't need anything more than this, in the base class if we want:

```python
def create(cls, **kwargs):
    return cls.objects.create(**kwargs)
```